lib/extension-es6.js

```diff
@@ -7,7 +7,8 @@ function es6(loader) {
 
   var transpiler, transpilerModule, transpilerRuntimeModule, transpilerRuntimeGlobal;
 
-  var isBrowser = typeof window != 'undefined';
+  var isBrowser = typeof window !== 'undefined' && !!window.Window && window instanceof window.Window;
+  var isWorker = typeof self !== 'undefined' && !!self.WorkerGlobalScope && self instanceof WorkerGlobalScope;
```

Comment:
OK, but then you might as well drop the check of `self` since `WorkerGlobalScope ` should cover that. Also `self` is defined with `self === window` in modern browsers anyways.

```js
var autoloadTranspiler = typeof window !== 'undefined' || typeof WorkerGlobalScope !== 'undefined';
```